src/configuration.ts

```diff
@@ -13,137 +12,187 @@ const DEBUG_PROXY_PORT = "debug_proxy_port";
-                return undefined;
-            }
-        }
-        throw e;
+async function getCloudConfigFromDbosCloud(appName: string, credentials: DbosCloudCredentials): Promise<Omit<CloudConfig, 'password'> | undefined> {
+  try {
+    const app = await getAppInfo(appName, credentials);
+    const db = await getDatabaseInfo(app.PostgresInstanceName, credentials);
+    const cloudConfig = {
+      host: db.HostName,
+      port: db.Port,
+      database: app.ApplicationDatabaseName,
+      user: db.DatabaseUsername,
+      appName: app.Name,
+    };
+    logger.debug("getCloudConfigFromVSCodeConfig", { appName, credentials, cloudConfig });
+    return cloudConfig;
+  } catch (e) {
+    if (isExecFileError(e)) {
+      if (e.stdout.includes("Error: not logged in") || e.stdout.includes("Error: Login expired")) {
+        return undefined;
+      }
     }
+    throw e;
+  }
 }
 
 function cfgString(value: string | undefined) {
-    return value?.length ?? 0 > 0 ? value : undefined;
+  return value?.length ?? 0 > 0 ? value : undefined;
 }
 
-function getCloudConfigFromVSCodeConfig(folder: vscode.WorkspaceFolder): CloudConfig {
-    const cfg = vscode.workspace.getConfiguration(TTDBG_CONFIG_SECTION, folder);
-
-    const host = cfg.get<string | undefined>(PROV_DB_HOST, undefined);
-    const port = cfg.get<number | undefined>(PROV_DB_PORT, undefined);
-    const database = cfg.get<string | undefined>(PROV_DB_DATABASE, undefined);
-    const user = cfg.get<string | undefined>(PROV_DB_USER, undefined);
+function getCloudConfigFromVSCodeConfig(folder: vscode.WorkspaceFolder): Partial<Omit<CloudConfig, 'password' | 'appName' | 'appId'>> {
+  const cfg = vscode.workspace.getConfiguration(TTDBG_CONFIG_SECTION, folder);
+
+  const host = cfg.get<string | undefined>(PROV_DB_HOST, undefined);
+  const port = cfg.get<number | undefined>(PROV_DB_PORT, undefined);
+  const database = cfg.get<string | undefined>(PROV_DB_DATABASE, undefined);
+  const user = cfg.get<string | undefined>(PROV_DB_USER, undefined);
+
+  const cloudConfig = {
+    host: cfgString(host),
+    port: port !== 0 ? port : undefined,
+    database: cfgString(database),
+    user: cfgString(user),
+  };
+  logger.debug("getCloudConfigFromVSCodeConfig", { folder: folder.uri.fsPath, cloudConfig });
+  return cloudConfig;
+}
 
-    return {
-        host: cfgString(host),
-        port: port !== 0 ? port : undefined,
-        database: cfgString(database),
-        user: cfgString(user),
-    };
+function domainSecretKey(domain: string) { return `dbos-ttdbg:domain:${domain}`; }
+function databaseSecretKey(db: Pick<CloudConfig, 'user' | 'host' | 'port' | 'database'> | DbosCloudDatabase) {
+  if ('host' in db) {
+    return `dbos-ttdbg:database:${db.user}@${db.host}:${db.port ?? 5432}/${db.database}`;
+  } else {
+    return `dbos-ttdbg:database:${db.DatabaseUsername}@${db.HostName}:${db.Port ?? 5432}/${db.PostgresInstanceName}`;
```

Comment:
Shouldn't be `db.PostgresInstanceName` here, because it should be the logical DB name (i.e., application database name `app.ApplicationDatabaseName`) when we connect to postgres.

Reply:
This function doesn't matter for now because we only have one DB user per postgres instance, but can be an issue in the future when we assign different user/password to different logical databases.

Reply:
As of now, we only call this method with cloud config, so I'll pull out the DbosCloudDatabase overload code. But I will double check to make sure I'm using the right DB name in the rest of the code